lib/web_ui/test/canvaskit/embedded_views_test.dart

```diff
@@ -640,14 +809,14 @@ void testMain() {
       result = diffViewList(<int>[1, 2, 3, 4], <int>[4, 3, 2, 1]);
       expect(result, isNotNull);
       expect(result!.viewsToAdd, <int>[3, 2, 1]);
-      expect(result.viewsToRemove, isEmpty);
+      expect(result.viewsToRemove, <int>[1, 2, 3]);
```

Comment:
Aha, so the logic of adding-without-removing (to move views) used to be in the computation of the diff, but it now is in the composition logic.

Reply:
Yes, I thought it was clearer if `diffViewList` didn't know about the optimization of not removing views that are going to be re-added, and so it was left to the compositor to handle that case